### PR TITLE
Allow states to declare _connection_ keys as documented

### PR DESCRIPTION
**Pull request by mchugh19**
### What does this PR do?
The state documentation refers to allowing _connection_user, password, and url keys but does not actually use them. This commit corrects that. 

### Previous Behavior
Connection information was only passed via salt pillar. 

### New Behavior
Connection information is now defined by the state if available. Otherwise, it falls back on the standard pillar.

### Tests written?
No

@jirikotlin
@mmoono - This may possibly conflict with some of your recent updates as well. If you'd like to re-push your changes we can see how to resolve.

### CODE REVIEW

**Comment by cachedout** [comment]
@mchugh19 Could you please look at these lint errors: https://jenkins.saltstack.com/job/PR/job/salt-pr-lint-n/8753/violations/file/salt/states/zabbix_host.py/

**Comment by mchugh19** [comment]
Will do. Currently traveling so it'll be next week.

File: salt/states/zabbix_host.py [hunk 1]
```diff
@@ -273,8 +283,15 @@ def absent(name):
                                    'new': 'Host {0} deleted.'.format(name),
                                    }
                             }
+    connection_args = {}
```
Comment (cachedout) [inline salt/states/zabbix_host.py]: This function doesn't accept `**kwargs`. Perhaps you just forgot to add it?

**Review by cachedout — CHANGES_REQUESTED** [review]

File: salt/states/zabbix_host.py (hunk 1 above)
Comment (mchugh19) [inline salt/states/zabbix_host.py]: Right you are. Corrected! Thanks!

**Pull request merged**